helm-charts/common/speecht5/templates/deployment.yaml

```diff
@@ -27,6 +27,35 @@ spec:
       {{- end }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- if not (hasPrefix "/data/" .Values.TTS_MODEL_PATH) }}
+      initContainers:
+        - name: model-downloader
+          envFrom:
+            - configMapRef:
+                name: {{ include "speecht5.fullname" . }}-config
+          securityContext:
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: true
+            capabilities:
+              drop:
+              - ALL
+              add:
+              - DAC_OVERRIDE
+              - FOWNER
+            seccompProfile:
+              type: RuntimeDefault
+          image: huggingface/downloader:0.17.3
+          command: ['sh', '-c']
+          args:
+            - |
+              huggingface-cli download --cache-dir /data --token $(HF_TOKEN)  {{ .Values.TTS_MODEL_PATH | quote }};
+              huggingface-cli download --cache-dir /data --token $(HF_TOKEN)  {{ .Values.VOCODER_MODEL| quote }};
+              chmod -R g+w /data/models--{{ replace "/" "--" .Values.TTS_MODEL_PATH }};
+              chmod -R g+w /data/models--{{ replace "/" "--" .Values.VOCODER_MODEL }}
```

Comment:
If model is already downloaded by init container, does _whole_ model dir still need to be writable by main container?  If main container still needs to update lock file there, make just the dir writable, not model files under it.

Btw. `initContainer` is now running as `root` group (not  just `root` user), so these `chmod` calls rely main container securityContext also to be using `root` group, and only its user ID being different.

To make that requirement explicit to users, I would suggest group ID to be set in pod `securityContext` with a comment stating why that group ID is there:
```
# Init container sets the downloaded model dir to be group writable, so that main
# container can keep its lock file there. This relies on containers using same group ID
```

Reply:
@eero-t the main container still needs the write permission in the model directory itself for some other reasons. I've tested tgi and confirmed that the write permission of model dir is needed.

Adding explicit groudID settings in securityContext is a good idea. I'll add that.